moncli/types.py

```diff
@@ -410,3 +410,37 @@ def _export(self, value):
         end  = value.end.strftime(DATE_FORMAT)
         return  {'week': {'startDate': start, 'endDate': end}}
 
+
+
+class TimelineType(MondayType):
+
+    native_type = en.cv.Timeline
+    allow_casts = (dict,)
+    null_value = {}
+
+    def _cast(self, value):
+        try:
+            from_date = value['from']
+            to_date = value['to']
+            return en.cv.Timeline(from_date=from_date, to_date=to_date)
+        except KeyError:
+            raise ConversionError('Could not convert value "{}" to Timeline.'.format(value))
+
+    def _set_metadate(self, value):
+        if value.settings['visualization_type']:
```

Comment:
It should be noted that this settings key does not appear if milestone is not set.  I would recommend wrapping 430 and 431 in a try/except on a KeyError.  If there is an error, run line 432.